recruit/types.go

```diff
@@ -0,0 +1,180 @@
+package recruit
+
+import (
+	"encoding/json"
+
+	zoho "github.com/schmorrison/Zoho"
+)
+
+type Error struct {
+	Code    string   `json:"code,omitempty"`
+	Details struct{} `json:"details,omitempty"`
+	Message string   `json:"message,omitempty"`
+	Status  string   `json:"status,omitempty"`
+}
+
+type PageInfo struct {
+	PerPage     int  `json:"per_page,omitempty"`
+	Count       int  `json:"count,omitempty"`
+	Page        int  `json:"page,omitempty"`
+	MoreRecords bool `json:"more_records,omitempty"`
+}
+
+type MultiSelect []string
+type Date = zoho.Date
+type Time = zoho.Time
+type Number int
+type Currency float64
+type Decimal float64
+type Percent float64
+type Long int64
+type Checkbox bool
+type URL string
+type Lookup struct {
+	Name string `json:"name,omitempty"`
+	ID   string `json:"id,omitempty"`
+}
+type Owner struct {
+	Name string `json:"name,omitempty"`
+	ID   string `json:"id,omitempty"`
+}
+type Layout struct {
+	Name string `json:"name,omitempty"`
+	ID   string `json:"id,omitempty"`
+}
+type AutoNumber string
+
+// SingleLine is the field type in Zoho that defines a single line input field
+type SingleLine string
+
+func (s *SingleLine) UnmarshalJSON(data []byte) error {
+	if string(data) == "null" {
+		t := SingleLine("")
+		s = &t
```

Comment:
Every location where you want to set the value of a pointer, you need to set like so:

Instead of:
```s = &t```

Use this:
```*s = t```

Reply:
done